### ISSUE TITLE
Authenticating a websocket connection via script

### ISSUE THREAD

**Issue by dian-workz** [miguelgrinberg/Flask-SocketIO#1335]
Hello,

I've been following some guides and tutorials on the web in setting up a Flask SocketIO server and using flask Login to authenticate. This seems to be working as expected if I log in and navigate to a protected REST endpoint. For this test I used something like Postman to connect to the Flask SocketIO server.

What I am trying to do now is to switch the **authenticated** user's HTTP session to a websocket connection.

Following [https://blog.miguelgrinberg.com/post/flask-socketio-and-the-user-session](https://blog.miguelgrinberg.com/post/flask-socketio-and-the-user-session)), I am attempting this using a POC Python client script to connect to the websocket server. The login REST endpoints are successfully called from the client script (in other words, this is not done via the browser) using socketio session, for example:

```python
with requests.Session() as session:
    resp = session.post(base + "/login", json={"email": "jdoe@acme.org", "password": "secret"})
```

but the client websocket connection gets rejected on the `@socketio.on('connect')` event stating that the `current_user.is_authenticated` is false when the websocket tries to connect.

So basically, my question are: 

1. Am I following the correct approach, and 
2. how does the authenticated user connect to the websocket using the same session?

Any advice is appreciated.

**Comment by miguelgrinberg** [comment]
The issue here is, I think, that when you log in using requests you get a cookie back, set by Flask-Login. Are you adding that cookie to the Socket.IO connection request? That's likely what's missing. You need to fish out the cookie out of the requests' cookie jar object, and then set it when you make the socket.io connection.

**Comment by dian-workz** [comment]
That seemed to have worked. Thank you very much for your quick reply.
